tests/unittests/GraphOptzTest.cpp

```diff
@@ -5451,6 +5451,31 @@ TEST_F(GraphOptz, EliminateSliceConcatTest) {
   checkNumericalEquivalence();
 }
 
+TEST_F(GraphOptz, EliminateSliceConcatWithReshapeTest) {
+  auto *src =
+      mod_.createPlaceholder(ElemKind::FloatTy, {4, 5, 3}, "src", false);
+  auto *A = F_->createSlice("A", src, {0, 0, 0}, {1, 5, 3});
+  auto *B = F_->createSlice("B", src, {1, 0, 0}, {2, 5, 3});
+  auto *C = F_->createSlice("C", src, {2, 0, 0}, {3, 5, 3});
+
+  auto *CN = F_->createConcat("Concat", {A, B, C}, 1);
+  F_->createSave("save", CN);
+
+  // src, A, B, C, CN.
+  EXPECT_EQ(F_->getNodes().size(), 5);
+
+  optimizedF_ = optimizeFunction(
+      F_, {FunctionPassID::EliminateSliceConcat, getDCEPassConfig()});
+
+  // src, FUSED_SLICE[A, B, C], RESHAPE, CN.
+  // FUSED_SLICE, RESHAPE are inserted by optimization.
+  EXPECT_EQ(optimizedF_->getNodes().size(), 4);
+
+  bindings_.allocate(src)->getHandle<float>().randomize(-10.0, 10.0,
+                                                        mod_.getPRNG());
+  checkNumericalEquivalence();
```

Comment:
You can specify the allowed error here -- let's make it 0.f since there's only data movement in this op.